lib/spack/spack/solver/asp.py

```diff
@@ -2710,8 +2710,7 @@ def setup(
         if env:
             dev_specs = tuple(
                 spack.spec.Spec(info["spec"]).constrained(
-                    "dev_path=%s"
-                    % spack.util.path.canonicalize_path(info["path"], default_wd=env.path)
+                    f'dev_path="{spack.util.path.canonicalize_path(info["path"], default_wd=env.path)}"'
```

Comment:
Should we move quoting internal to `canonicalize_path`?

Reply:
No, I think that adding it to `canonicalize_path` will cause problems everywhere that we use it *outside* of a spec constructor.